spline-testing/jmetered-backed-simple/run_jmetered_spline.py

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env python3
+import argparse
+import subprocess
+import docker
+import os
+import platform
+import pandas as pd
+import matplotlib.pyplot as plt
+
+GIT_CLONE_SPLINE = "git clone https://github.com/AbsaOSS/spline.git"
+GIT_CHEKOUT_SPLINE_BRANCH = "git checkout {branch}"
+SPLINE_DEFAULT_BRANCH = "develop"
+
+SPLINE_CORE_VERSION = "1.0.0-SNAPSHOT"  # needs to be in sync with .env
+CUSTOM_IMAGES = [f"testing-spline-db-admin:{SPLINE_CORE_VERSION}", f"testing-spline-rest-server:{SPLINE_CORE_VERSION}"]
+
+SPLINE_BUILD = "{mvn} install -DskipTests"
+
+JMETER_COLNAME_TIMESTAMP = "timeStamp"
+JMETER_COLNAME_ELAPSED= "elapsed"
+JMETER_COLNAME_LABEL = "label"
+# coming from Jmeter, but custom-added (graphType,operationCount,attributeCount,readCount)
+JMETER_COLNAME_GRAPH_TYPE = "graphType"
+JMETER_COLNAME_OP_COUNT = "operationCount"
+JMETER_COLNAME_ATTR_COUNT = "attributeCount"
+JMETER_COLNAME_READ_COUNT = "readCount"
+
+# script folders:
+RESULTS_FOLDER_NAME = "results"
+REFERENCE_FOLDER_NAME = "reference"
+PROCESSED_FOLDER_NAME = "processed_results"
+GRAPHS_FOLDER_NAME = "graphs"
+
+# Matplotlib:
+PLOT_MARKER = "o"
+PLOT_MARKER_SIZE = 2
+PLOT_DPI = 300
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        prog='run_jmetered_spline',
+        description='Spline backend simple testing (jmetered)',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter  # prints default values, too, on help (-h)
+    )
+
+    parser.add_argument('-n', '--no-rebuild', action='store_true', dest="no_rebuild", default=False,
+                        help="if specified, spline will not be rebuilt (use when rerunning on an existing codebase)")
+    # parser.add_argument('-v', '--verbose', action="store_true", default=False,
+    #                     help="prints extra information while running.")
+
+    parser.add_argument('-b', '--spline-branch', dest="spline_branch", default=SPLINE_DEFAULT_BRANCH,
+                        help="Name of spline branch to test")
+
+    return parser.parse_args()
+
+
+def get_mvn_by_os() -> str:
+    if platform.system() == "Windows":
+        return "mvn.cmd"
+    else:
+        return "mvn"
+
+
+def build_spline(branch):
+    print(f"Getting spline (via: '{GIT_CLONE_SPLINE}')")
+    subprocess.run(GIT_CLONE_SPLINE)
```

Comment:
From the [subprocess doc](https://docs.python.org/3/library/subprocess.html)

> args is required for all calls and should be a string, or a sequence of program arguments. Providing a sequence of arguments is generally preferred, as it allows the module to take care of any required escaping and quoting of arguments (e.g. to permit spaces in file names). **If passing a single string, either shell must be [True](https://docs.python.org/3/library/constants.html#True) (see below) or else the string must simply name the program to be executed without specifying any arguments.**

```suggestion
    subprocess.run(GIT_CLONE_SPLINE, shell=True)
```

Reply:
Thanks, applied.